### ISSUE TITLE
Feature Request: bi-directional Signal integration

### ISSUE THREAD

**Issue by Lucky225** [guardianproject/haven#174]
Why require an onion server when you could just send commands back to Haven through Signal to get what you want? Not only to retrieve all alerts (say you lost internet yourself and never got the alert on SIgnal, you could ask it to resend), but you could also have a ping command instead of a heartbeat if desired to make sure the thing is still alive as well take photos/audio on the fly, get GPS if the thing starts moving, etc.  require password security if desired to make sure if your signal gets hijacked someone doesn't start receiving your notifications, etc. etc.

**Comment by leonardoporpora** [comment]
Guys we have to remember about SS7 Vulnerability this could give the access to someone else, hackers and governments that could disable remotely Haven. I would never enable that feature!

**Comment by Lucky225** [comment]
Signal doesn't use SS7, but that's why I suggested there be a passcode in case your Signal does get hijacked, and further there should be a way to check safety numbers from both the haven device and the remote device. Hence feature request :P

**Comment by leonardoporpora** [comment]
When you receive the verification code is an SMS so it’s vulnerable to SS7 and when someone has got the access to Signal as you required he could disable Haven remotely...

**Comment by Lucky225** [comment]
You're still not comprehending. Yes signal verification code is sent via
regular SMS and can therefore be used to hijack a signal user, however if
your signal gets hijacked the safety numbers change, therefore the app
could protect against this if safety numbers were verifiable. Secondly, if
a passcode was set for remote commands, in addition to safety number
checks, no just jacking a signal user would not let you disable Haven.

**Comment by leonardoporpora** [comment]
Right now there is no way to verify Signal Secret Numbers. So the only way to protect is the passcode.

**Comment by Lucky225** [comment]
There's only no way to verify because the app doesn't currently offer it...
hence again, feature request :) I didnt initially include that as I saw it
already acknowledged in issue #82... and now I see my basic request was
already acknowledged in #52 whoops :X

On Dec 31, 2017 8:15 PM, "leonardoporpora" <notifications@github.com> wrote:

Right now there is no way to verify Signal Secret Numbers. So the only way
to protect is the passcode.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/guardianproject/haven/issues/174#issuecomment-354634659>,
or mute the thread

**Comment by leonardoporpora** [comment]
Yes I’m one of the people that sent the request to implement verification of Signal Secret numbers
